Small typo in allowed args for orphanpoolexpiry

ptschip opened the pull request.

ptschip committed:
Small typo in allowed args for orphanpoolexpiry

awemany reviewed (commented):
Seems to be a common typo:
```
src/txdb.h:25:static const int64_t nDefaultDbCache = 300; // BU Xtreme Thinblocks bump to support a larger ophan cache

src/qt/rpcconsole.h:94:    /** Set number of transactions in ophan pool in the UI */
```
Maybe this is also the right time to complain that there's no test for that command line flag... :)

ptschip commented:
all the orphan pool tests are in DOS_tests.cpp, probably not the best
place for them but they are fully covered i believe.

On 30/05/2017 10:34 AM, awemany wrote:
>
> *@awemany* commented on this pull request.
>
> Seems to be a common typo:
>
> |src/txdb.h:25:static const int64_t nDefaultDbCache = 300; // BU Xtreme
> Thinblocks bump to support a larger ophan cache
> src/qt/rpcconsole.h:94: /** Set number of transactions in ophan pool
> in the UI */ |
>
> Maybe this is also the right time to complain that there's no test for
> that command line flag... :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/633#pullrequestreview-41020687>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMRF0HtNpkJetgOC9yuJjy8WewSHMQviks5r_FM7gaJpZM4NqlXU>.
>

ptschip committed:
Add DbgAssert for nBytesOrphanPool

ptschip committed:
Use DbgAssert to ensure that there is an orphan before continuing

When processing mapOrphanTransactionByPrev make sure there is a matching
orphan in the orphan pool before we continue.  This should never happen
because when we erase orphans we also erase OrphanTransactionsByPrev but
we include this as an added safeguard.
